feat(ProfileShowcase): Create dirty state component for the web tab and integrate the UI

alexjba opened the pull request:
### What does the PR do

Closing #13495

#### Note
This will be merged after https://github.com/status-im/status-desktop/issues/13799 to align the save action as well.

<!-- Fill in the relevant information below to help us evaluate your proposed changes. -->

Changes:
1. Create the dirty state component for the web tab
2. Update the ProfileSocialLinksPanel to support the new model
3. Integrate the ProfileSocialLinksPanel actions with the dirty state component
4. Update storybook
5. Quick fix here ui/StatusQ/src/StatusQ/Controls/StatusSeedPhraseInput.qml because the `Popups` component cannot be used in storybook with qt 5.15.8

### Affected areas

Profile showcase
<!-- List the affected areas (e.g wallet, browser, etc..) -->

### Screenshot of functionality (including design for comparison)

https://github.com/status-im/status-desktop/assets/47811206/868e9a5e-dacb-4250-ab93-064d29d9be1f

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 7ef54cf4 | [#1](https://ci.status.im/job/status-desktop/job/prs/job/macos/job/aarch64/job/package/job/PR-13900/1/) | 2024-03-08 09:40:29 | ~6 min | `macos/aarch64` | [:apple:`dmg`](https://status-im-desktop-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Desktop-240308-093401-7ef54c-pr13900-aarch64.dmg) |

noeliaSD commented on ui/app/AppLayouts/Profile/helpers/ProfileShowcaseDirtyState.qml:
```diff
@@ -16,9 +16,18 @@ import utils 1.0
+      * model is part of the source model. False if the showcase model is a
+      * separate model.
+      */
+    property bool singleModelMode: !joined.rightModel
```
readonly?

alexjba replied:
no, it's an input property. This is just the default property

noeliaSD commented on ui/imports/utils/ProfileUtils.qml:
```diff
@@ -78,6 +78,16 @@ QtObject {
         return Constants.socialLinkType.custom
     }
 
+    function linkTypeToIcon(linkType) {
```
Have you seen the icon also comes from NIM? With this new function, could we remove the code there? FYI: https://github.com/status-im/status-desktop/issues/13817

alexjba replied:
yeah. The nim icons need to be removed

noeliaSD reviewed (commented):
LGTM!!

I've just left some comments and a minor issue found on the UI:
<img width="1193" alt="Screenshot 2024-03-08 at 12 55 27" src="https://github.com/status-im/status-desktop/assets/97019400/f30901fb-26a3-42e9-80ff-4bed8155c737">

alexjba commented:
@MishkaRogachev I've integrated the new `save` as well in the web tab, but it seems it's not working. The async task logs shows that the social links updates are ok, but nothing is saved.

alexjba commented on storybook/pages/ProfileSocialLinksPanelPage.qml:
```diff
@@ -90,43 +76,25 @@ SplitView {
-                    logs.logEvent("ProfileStore::saveSocialLinks", ["silent"], arguments)
-                }
+
+            onAddSocialLink: {
```
fixed

alexjba commented:
This PR is ready for merge now. Save work well for the web tab as well thanks to @MishkaRogachev's commit integrated here.

alexjba commented:
> > This PR is ready for merge now. Save work well for the web tab as well thanks to @MishkaRogachev's commit integrated here.
> 
> I guess it should be properly rebased; it contains a commit from Mishka that's been merged to master meanwhile

It's been merged here. I've updated the model adapter as well to remove the unnecessary code now that we have the light models.

caybro commented:
> > > This PR is ready for merge now. Save work well for the web tab as well thanks to @MishkaRogachev's commit integrated here.
> > 
> > I guess it should be properly rebased; it contains a commit from Mishka that's been merged to master meanwhile
> 
> It's been merged here. I've updated the model adapter as well to remove the unnecessary code now that we have the light models.

It still has a merge conflict

alexjba commented:
> > > > This PR is ready for merge now. Save work well for the web tab as well thanks to @MishkaRogachev's commit integrated here.
> > > 
> > > 
> > > I guess it should be properly rebased; it contains a commit from Mishka that's been merged to master meanwhile
> > 
> > 
> > It's been merged here. I've updated the model adapter as well to remove the unnecessary code now that we have the light models.
> 
> It still has a merge conflict

fixed

caybro reviewed (commented):
LGTM otherwise!

MishkaRogachev reviewed (approved):
Looks good!

alexjba commented on ui/app/AppLayouts/Profile/helpers/VisibilityAndPositionDirtyStateModel.qml:
```diff
@@ -75,9 +75,14 @@ WritableProxyModel {
-        function getVisibleEntries() {
-            const roles = ["showcaseKey", "showcasePosition", "showcaseVisibility"]
-            const keysAndPos = ModelUtils.modelToArray(root, roles)
+        function getVisibleEntries(roleNames = ["showcaseKey", "showcasePosition", "showcaseVisibility"]) {
```
Any type of binding I do in the SFPM `name` seems to be evaluated too late and the join or other model operations are failing. I think we can keep it like this for now.

I can add a task to further analyse if you think it's worth
